### PR TITLE
TMA-299: Data Product

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-examples-opensource-repository-check/3/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-examples-tests-zuul/36/ : SUCCESS in 1m 17s

File: book_src/13_working_with_lifecycle/02_data_products.asciidoc.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+=== Working with data products
+by Jakub Mahnert
+
+==== Problem
+You would like to encapsule multiple segments into a data product for better granularity.
+
+==== Solution
+
+This solution is based on the terminology and examples presented in chapter regarding Segments.
```
Comment (panjan) [inline book_src/13_working_with_lifecycle/02_data_products.asciidoc.erb]: Would be nice to have a link to the Segments chapter.

File: book_src/src/13_working_with_lifecycle/data_product.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+master_project = client.create_project(title: 'master-project', auth_token: 'auth-token')
+# Create new lifecycle segment without referencing a data product
+first_segment = domain.create_segment(segment_id: 'segment-1', master_project: project)
+# the API defaults the data product to 'default', which should be present in all domains
+# the above call is equal to
+default_data_product = domain.data_products('default')
+default_data_product.create_segment(segment_id: 'segment-1', master_project: project)
+# Create another segment as a part of the data product
+second_segment = domain.create_segment(segment_id: 'segment-2', master_project: project)
+#
```
Comment (panjan) [inline book_src/src/13_working_with_lifecycle/data_product.rb]: empty comment

File: book_src/src/13_working_with_lifecycle/data_product.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require 'gooddata'
```
Comment (panjan) [inline book_src/src/13_working_with_lifecycle/data_product.rb]: Given we don't s explain where e.g. `domain` comes from - is it necessary to `require 'gooddata'`.

File: book_src/13_working_with_lifecycle/01_wip_segments.asciidoc.erb [hunk 4]
```diff
@@ -28,4 +28,4 @@ Having defined some terminology we present code that will do several things.
 - adds a client
```
Comment (panjan) [inline book_src/13_working_with_lifecycle/01_wip_segments.asciidoc.erb]: Please regenerate the docs using rake.

File: book_src/13_working_with_lifecycle/02_data_products.asciidoc.erb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+This code will do several things:
+
+- sets up a segment in the legacy way, referencing no dataproduct
+- sets up another segment as a part of a dataproduct
+- shows that the segments are only included in the specific dataproduct that was used to create them
```
Comment (panjan) [inline book_src/13_working_with_lifecycle/02_data_products.asciidoc.erb]: Where do we show this?

File: book_src/13_working_with_lifecycle/02_data_products.asciidoc.erb [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+<%= render_ruby 'src/13_working_with_lifecycle/segments.rb' %>
```
Comment (panjan) [inline book_src/13_working_with_lifecycle/02_data_products.asciidoc.erb]: Is the path correct?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-examples-opensource-repository-check/4/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-examples-tests-zuul/37/ : SUCCESS in 19s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-examples-opensource-repository-check/5/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-examples-tests-zuul/38/ : SUCCESS in 19s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-examples-opensource-repository-check/6/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-examples-tests-zuul/39/ : SUCCESS in 22s

**Comment by panjan** [comment]
What do you think @Teknikov?

**Comment by Teknikov** [comment]
Cookbook text looks fine to me. Cheers.

**Pull request merged**